pkg/apis/v1alpha1/serviceimport.go

```diff
@@ -141,3 +147,46 @@ type ServiceImportList struct {
 	// +listType=set
 	Items []ServiceImport `json:"items"`
 }
+
+const (
+	// ServiceImportConditionReady is true when the Service Import is ready.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Ready"
+	// * "PartialIPFamilyCompatibility"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Progressing"
+	// * "PartialIPFamilyCompatibility"
+	// * "IPFamilyNotSupported"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Progressing"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceImportConditionReady = "Ready"
+
+	// ServiceImportReasonReady is used with the "Ready" condition when the
+	// condition is True.
+	ServiceImportReasonReady = "Ready"
+
+	// ServiceExportReasonPartialIPFamilyCompatibility is used with the "Ready"
+	// condition when the local cluster partially supports the imported service
+	// IP families.
+	ServiceExportReasonPartialIPFamilyCompatibility = "PartialIPFamilyCompatibility"
```

Comment:
Should this be set when the importing cluster offers _more_ IP families than are available for endpoints (as a warning that in a dual-stack cluster, an IPv6 only client may fail to connect to a ServiceImport with only IPv4 backends available)?

I think ^ makes more sense than the inverse (endpoints are available for IPFamilies other than those supported by the importing cluster, but at least some are available for a matching IPFamily (because in this case, all clients can route to a backend and the ServiceImport could for all practical purposes be considered `Ready`)

Reply:
This was the for the inverse but I wasn't that convinced either when I wrote that so you writing about that it should be rather `Ready` comfort me into dropping this actually :sweat_smile:.

> an IPv6 only client may fail to connect to a ServiceImport with only IPv4 backends available)?

I don't think that this can happen because it's only the Service IPFamily that you can select in a dual stack cluster, IIRC all the pods are pretty much required to be dual stack.